### PR TITLE
ensure that CrateDocCollector always releases the searchcontext upon …

### PR DESCRIPTION
**Pull request by dobe**
…kill

### CODE REVIEW

**Comment by dobe** [comment]
@matriv pls review

File: sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public void testKillCollectorKill() throws Exception {
+
+        CrateSearchContext sc = mock(CrateSearchContext.class, Answers.RETURNS_MOCKS.get());
+        RowReceiver rowReceiver = mock(RowReceiver.class, Answers.RETURNS_MOCKS.get());
+        CrateDocCollector c = new CrateDocCollector(sc, MoreExecutors.directExecutor(), false, null,
+            rowReceiver, ImmutableList.<Input<?>>of(), ImmutableList.<LuceneCollectorExpression<?>>of());
+
+        c.kill(null);
+
+        verify(sc, atLeastOnce()).clearReleasables(any(SearchContext.Lifetime.class));
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java]: Why is this: atLeastOnce? and not once?

File: sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.mockito.Answers;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+public class CrateDocCollectorTest {
+
+    @Test
+    public void testKillCollectorKill() throws Exception {
+
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java]: empty line

File: sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        CrateSearchContext sc = mock(CrateSearchContext.class, Answers.RETURNS_MOCKS.get());
+        RowReceiver rowReceiver = mock(RowReceiver.class, Answers.RETURNS_MOCKS.get());
+        CrateDocCollector c = new CrateDocCollector(sc, MoreExecutors.directExecutor(), false, null,
+            rowReceiver, ImmutableList.<Input<?>>of(), ImmutableList.<LuceneCollectorExpression<?>>of());
+
+        c.kill(null);
+
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java]: empty line

File: sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        verify(sc, atLeastOnce()).clearReleasables(any(SearchContext.Lifetime.class));
+        verify(rowReceiver, only()).kill(any(Throwable.class));
+
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java]: empty line

File: sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/collectors/CrateDocCollectorTest.java]: empty line

**Pull request merged**